apis/Google.Cloud.Datastore.V1/Google.Cloud.Datastore.V1/EmulatorDetection.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+namespace Google.Cloud.Datastore.V1
+{
+    /// <summary>
+    /// Specifies how the use of the Datastore emulator is detected via environment variables,
+    /// when used by <see cref="DatastoreDbBuilder"/>.
+    /// </summary>
+    public enum EmulatorDetection
+    {
+        /// <summary>
+        /// Ignore environment variables entirely.
+        /// </summary>
+        None = 0,
+
+        /// <summary>
+        /// Always connect to the production servers, but throw an exception if
+        /// environment variables exist that would suggest the emulator is expected.
+        /// </summary>
+        ProductionOnlyThrowOnEmulatorEnvironment = 1,
```

Comment:
I slightly dislike the inconsistent naming between this and `EnumatorOnly`.
Either they should both have the `ThrowOn...` suffix or neither should.

Reply:
Hmm. Will make it just ProductionOnly for now. We have a while before this will go GA, and should probably do another review of the API before then anyway.